lib/sshutils/sshutils.go

```diff
@@ -296,41 +318,63 @@ func (s *SSHUtilImpl) CopyFileToRemoteConcurrently(sshConfig SSHConfig, srcFileP
 	return results
 }
 
-func (s *SSHUtilImpl) CreateKnownHosts() error {
-	f, fErr := os.OpenFile(filepath.Join(os.Getenv("HOME"), ".ssh", "known_hosts"), os.O_CREATE, 0600)
+func (s *SSHUtilImpl) CreateKnownHosts(knownHostPath string) error {
+	f, fErr := os.OpenFile(knownHostPath, os.O_CREATE, 0600)
 	if fErr != nil {
 		return fErr
 	}
 	f.Close()
 	return nil
 }
 
-func (s *SSHUtilImpl) CheckKnownHosts() (ssh.HostKeyCallback, error) {
-	err := s.CreateKnownHosts()
+func (s *SSHUtilImpl) CheckKnownHosts(knownHostPath string) (ssh.HostKeyCallback, error) {
+	err := s.CreateKnownHosts(knownHostPath)
 	if err != nil {
 		return nil, err
 	}
-	kh, e := s.SshClient.New(filepath.Join(os.Getenv("HOME"), ".ssh", "known_hosts"))
+	kh, e := s.SshClient.New(knownHostPath)
 	if e != nil {
 		s.logger.Error("Error while getting the known host: %v", e)
 		return nil, e
 	}
 	return kh, nil
 }
 
-func (s *SSHUtilImpl) AddHostKey(host string, remote net.Addr, pubKey ssh.PublicKey) error {
-	// add host key if host is not found in known_hosts, error object is return, if nil then connection proceeds,
+func getLastLine(input string) string {
```

Comment:
Was it intended to implement this `getLastLine` method here while there is one in `stringutils.GetLastLine` that is being called by the functions in this file?

Reply:
no this is a miss from my end. I should remove it from here as I have already moved it to stringutils. Thanks for pointing, I have removed it